kong/api/crud_helpers.lua

```diff
@@ -128,23 +138,38 @@ function _M.paginated_set(self, dao_collection)
 
   local next_url
   if offset then
-    offset = ngx.encode_base64(offset)
+    offset = encode_base64(offset)
     next_url = self:build_url(self.req.parsed_url.path, {
-      port = self.req.parsed_url.port,
-      query = ngx.encode_args {
+      port     = self.req.parsed_url.port,
+      query    = encode_args {
         offset = offset,
-        size = size
+        size   = size
       }
     })
   end
 
-  return responses.send_HTTP_OK {
+  local data
+
+  if #rows == 0 then
     -- FIXME: remove and stick to previous `empty_array_mt` metatable
     -- assignment once https://github.com/openresty/lua-cjson/pull/16
     -- is included in the OpenResty release we use.
-    data = #rows > 0 and rows or cjson.empty_array,
-    total = total_count,
-    offset = offset,
+    data = cjson.empty_array
+
+  else
+    data = rows
+
+    if type(cb) == "function" then
+      for _, row in ipairs(rows) do
+        cb(row)
+      end
```

Comment:
I think sometimes replacements are fine. E.g. you want to delete 5 columns and have just 2. Then it is easier to return a new table with these two columns that set 5 columns in original to `nil`. But I think in general you usually want to decode some field, or add new fields (e.g. fields counted from other fields, like full name could be concatenated from first name and last name).